core/src/main/java/io/questdb/cutlass/line/udp/LineUdpLexer.java

```diff
@@ -367,21 +384,13 @@ public CharSink put(char c) {
             return this;
         }
 
-        @Override
-        public CharSink put(char[] chars, int start, int len) {
-            throw new UnsupportedOperationException();
-        }
-
         private void extend() {
             int capacity = ((int) (bufferHi - buffer) * 2);
             if (capacity < 0) {
                 // can't realistically reach this in test :(
                 throw LineProtoException.INSTANCE;
             }
-            long buf = Unsafe.malloc(capacity);
-            Unsafe.getUnsafe().copyMemory(buffer, buf, (dstPos - buffer));
-            Unsafe.free(buffer, bufferHi - buffer);
-
+            long buf = Unsafe.realloc(buffer, bufferHi - buffer, capacity, MemoryTag.NATIVE_DEFAULT);
             long offset = dstTop - buffer;
             bufferHi = buf + capacity;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).